python_modules/dagster/docs/reference.rst

```diff
@@ -0,0 +1,212 @@
+
+DAG
+^^^
+
+DAG is short for `directed acyclic graph`. In this context, we are concerned with graphs where the
+nodes are computations and the edges are dependencies between those computations. The dependencies
+are `directed` because the outputs of one computation are the inputs to another.
+These graphs are `acyclic` because there are no circular dependencies -- in other words, the graph
+has a clear beginning and end, and we can always figure out what order to execute its nodes in.
+
+Execution Plan
+^^^^^^^^^^^^^^
+An execution plan is a concrete plan for executing a DAG of execution steps created by compiling a
+pipeline and a config. The execution plan is aware of the topological ordering of the execution
+steps, enabling physical execution on one of the available executor engines (e.g., in-process,
+multiprocess, using Airflow).
+
+Users do not directly instantiate or manipulate execution plans.
+
+Execution Step
+^^^^^^^^^^^^^^
+
+Execution steps are concrete computations, one or more of which corresponds to a solid in a pipeline
+that has been compiled with a config. Some execution steps are generated in order to compute the
+core transform functions of solids, but execution steps may also be generated in order to
+materialize outputs, check expectations against outputs, etc.
+
+Users do not directly instantiate or manipulate execution steps.
+
+Dagster Event
+^^^^^^^^^^^^^
+
+When a pipeline is executed, a stream of events communicate the progress of its execution. This
+includes top level events when the pipeline starts and completes, when execution steps succeed,
+fail, or are skipped due to upstream failures, and when outputs are generated and materialized.
+
+Users do not directly instantiate or manipulate Dagster events, but they are consumed by the GraphQL
+interface that supports the Dagit tool.
+
+InputDefinition
+^^^^^^^^^^^^^^^
+
+Optionally typed definition of the data that a solid requires in order to execute. Defined inputs
+may often also be shimmed through config. Inputs are defined using the
+:class:`InputDefinition <dagster.InputDefinition>` class, usually when defining a solid.
+
+OutputDefinition
+^^^^^^^^^^^^^^^^
+
+Optionally typed definition of the result that a solid will produce. Outputs are defined using the
+:class:`OutputDefinition <dagster.OutputDefinition>` class, usually when defining a solid.
+
+Dagster Types
+^^^^^^^^^^^^^
+
+The Dagster type system allows authors of solids and pipelines to optionally and gradually define
+the types of the data that flows between solids, and so to introduce compile-time and runtime checks
+into their pipelines.
+
+Types also allow for custom materialization, and are typically defined using the
+:func:`@dagster_type <dagster.dagster_type>` decorator or the
+:func:`as_dagster_type <dagster.as_dagster_type>` API. It is also possible to inherit from
+:class:`RuntimeType <dagster.RuntimeType>` directly.
+
+Resources
```

Comment:
this should be closer to pipeline, maybe next to an explicit section on context

Reply:
yep